Neos.ContentRepository/Classes/Eel/FlowQueryOperations/CanEvaluateNodeContextTrait.php

```diff
@@ -0,0 +1,46 @@
+<?php
+declare(strict_types=1);
+
+namespace Neos\ContentRepository\Eel\FlowQueryOperations;
+
+use Neos\ContentRepository\Domain\Projection\Content\NodeInterface;
+use Neos\ContentRepository\Domain\Projection\Content\TraversableNodeInterface;
+
+trait CanEvaluateNodeContextTrait
+{
+    /**
+     * {@inheritdoc}
+     *
+     * @param array|\Traversable (or array-like object) $context onto which this operation should be applied
+     */
+    public function checkContextForNodeInterface($context): bool
+    {
+        if (is_array($context)) {
+            return count($context) > 0 ? reset($context) instanceof NodeInterface : true;
+        } elseif ($context instanceof \Traversable) {
+            foreach ($context as $item) {
+                return $item instanceof NodeInterface;
+            }
+            return true;
```

Comment:
i dont think we need this line.
it would be for the case that $context has no iterations, and in this case why should it pass?